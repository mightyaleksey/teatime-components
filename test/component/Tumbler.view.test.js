'use strict';

const Tumbler = require('../../component/Tumbler');
const React = require('react');
const renderer = require('react-test-renderer');

test('defaultProp #size should have a value `m`', () => {
  expect(Tumbler.defaultProps.size).toBe('m');
});

test('Renders correctly with default props', () => {
  const tree = renderer.create(
    <Tumbler
      name='control'>
      Notifications
    </Tumbler>
  );

  expect(tree).toMatchSnapshot();
});

test('Renders correctly with defined props', () => {
  const tree = renderer.create(
    <Tumbler
      checked
      className='outer'
      disabled
      hasLabel={false}
      id='controls-id'
      name='control'
      offText='~~OFF~~'
      onText='~~ON~~'>
      Notifications
    </Tumbler>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
