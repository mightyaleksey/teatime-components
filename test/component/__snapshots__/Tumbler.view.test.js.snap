// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly with default props 1`] = `
<span
  className="wrapper"
  onChange={undefined}
>
  <span
    className="baseline"
  >
    <div
      className="shape"
    >
      <input
        checked={undefined}
        className="native"
        defaultChecked={undefined}
        disabled={undefined}
        id="_teatime1"
        name="control"
        onBlur={undefined}
        onChange={[Function]}
        onFocus={undefined}
        type="checkbox"
        value={undefined}
      />
      <label
        className="control"
        htmlFor="_teatime1"
      >
        <span
          className="label"
        >
          On
        </span>
        <span
          className="label"
        >
          Off
        </span>
        <span
          className="delimiter"
        >
           
        </span>
      </label>
    </div>
  </span>
</span>
`;

exports[`Renders correctly with defined props 1`] = `
<span
  className="wrapper outer"
  onChange={undefined}
>
  <span
    className="baseline"
  >
    <div
      className="shape"
    >
      <input
        checked={true}
        className="native"
        defaultChecked={undefined}
        disabled={true}
        id="controls-id"
        name="control"
        onBlur={undefined}
        onChange={[Function]}
        onFocus={undefined}
        type="checkbox"
        value={undefined}
      />
      <label
        className="control"
        htmlFor="controls-id"
      >
        <span
          className="label"
        >
          ~~ON~~
        </span>
        <span
          className="label"
        >
          ~~OFF~~
        </span>
        <span
          className="delimiter"
        >
           
        </span>
      </label>
    </div>
  </span>
</span>
`;
